<launch>
	<group ns="bumper_laser">
		<node name="range_filter" pkg="laser_filters" type="scan_to_cloud_filter_chain">
			<rosparam command="load" file="$(find kut_ugv_lasers)/config/bumper_laser_filters.yaml" />
		</node>
		<node name="cloud_assembler" pkg="laser_assembler" type="point_cloud2_assembler">
			<remap from="cloud" to="cloud_filtered" />
			<param name="max_clouds" value="150" />
			<param name="fixed_frame" value="/odom" />
		</node>
		<node name="periodic_cloud_publisher" pkg="kut_ugv_lasers" type="publish_assembled_cloud">
			<param name="publish_rate" value="25.0" />
			<param name="distance_horizon" value="30.0" />
			<remap from="vehicle_state" to="/vehicle/state" />
		</node>
	</group>

	<group ns="roof_laser">
		<node name="range_filter" pkg="laser_filters" type="scan_to_cloud_filter_chain" output="screen">
			<rosparam command="load" file="$(find kut_ugv_lasers)/config/roof_laser_filters.yaml" />
		</node>
		<node name="cloud_assembler" pkg="laser_assembler" type="point_cloud2_assembler" output="screen">
			<remap from="cloud" to="cloud_filtered" />
			<param name="max_clouds" value="1200" />
			<param name="fixed_frame" value="/odom" />
		</node>
		<node name="periodic_cloud_publisher" pkg="kut_ugv_lasers" type="publish_assembled_cloud">
			<param name="publish_rate" value="25.0" />
			<param name="distance_horizon" value="100.0" />
			<remap from="vehicle_state" to="/vehicle/state" />
		</node>
	</group>

</launch>
